core/src/main/kotlin/io/customer/sdk/core/extensions/ContextExtensions.kt

```diff
@@ -0,0 +1,29 @@
+package io.customer.sdk.core.extensions
+
+import android.content.Context
+import android.content.pm.ApplicationInfo
+import android.content.pm.PackageManager
+import android.os.Build
+import android.os.Bundle
+import io.customer.sdk.core.di.SDKComponent
+
+/**
+ * Gets meta-data from AndroidManifest.xml file.
+ *
+ * @return The meta-data bundle from application info.
+ */
+fun Context.applicationMetaData(): Bundle? = runCatching {
+    val appInfo: ApplicationInfo = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+        packageManager.getApplicationInfo(
+            packageName,
+            PackageManager.ApplicationInfoFlags.of(PackageManager.GET_META_DATA.toLong())
+        )
+    } else {
+        @Suppress("DEPRECATION")
+        packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
+    }
+
+    return@runCatching appInfo.metaData
+}.onFailure { ex ->
+    SDKComponent.logger.error("Failed to get application meta-data: ${ex.message}")
+}.getOrNull()
```

Comment:
```suggestion
/**
 * Gets meta-data from AndroidManifest.xml file.
 *
 * @return The meta-data bundle from application info, or null if retrieval fails.
 */
fun Context.getApplicationMetaData(): Bundle? {
    return try {
        val appInfo = getApplicationInfo()
        appInfo.metaData
    } catch (e: PackageManager.NameNotFoundException) {
        SDKComponent.logger.error("Failed to get application meta-data: ${e.message}")
        null
    }
}

/**
 * Helper function to get ApplicationInfo based on Android version.
 */
@Suppress("DEPRECATION")
private fun Context.getApplicationInfo(): ApplicationInfo {
    return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
        packageManager.getApplicationInfo(
            packageName,
            PackageManager.ApplicationInfoFlags.of(PackageManager.GET_META_DATA.toLong())
        )
    } else {
        packageManager.getApplicationInfo(packageName, PackageManager.GET_META_DATA)
    }
}
```

I think this method is doing too much 😅, we can probably decouple it to make it more readable and keep it fail-proof?

Reply:
This is a bit tricky because the messy code is mainly due to deprecation warnings and fetching application info the recommended way for different API levels, while try-catch is for application info not the meta-data. I’ve tried splitting it into smaller method that throws, breaking it into two halves. Let me know if this looks better.